notification-service/src/main/java/org/hypertrace/alert/engine/notification/service/NotificationEventProcessor.java

```diff
@@ -42,14 +42,20 @@
         .collect(Collectors.toMap(NotificationChannel::getChannelId, Function.identity()));
   }
 
-  void process(NotificationEvent notificationEvent) throws IOException {
+  public void process(NotificationEvent notificationEvent) {
     EventRecord eventRecord = notificationEvent.getEventRecord();
     if (!eventRecord.getEventType().equals(METRIC_ANOMALY_ACTION_EVENT_TYPE)) {
       LOGGER.debug("Received unsupported event type {}", eventRecord.getEventType());
       return;
     }
-    MetricAnomalyNotificationEvent metricAnomalyNotificationEvent =
-        MetricAnomalyNotificationEvent.fromByteBuffer(eventRecord.getEventValue());
+
+    MetricAnomalyNotificationEvent metricAnomalyNotificationEvent;
+    try {
+      metricAnomalyNotificationEvent =
+          MetricAnomalyNotificationEvent.fromByteBuffer(eventRecord.getEventValue());
+    } catch (IOException e) {
+      throw new RuntimeException("Exception deserializing MetricAnomalyNotificationEvent", e);
```

Comment:
Shouldn't this be IOException only as its serde failure?

Reply:
Wrapping in an unchecked exception

Reply:
But, why?

Reply:
Take care of it in follow-up.